website/src/controllers/groups.js

```diff
@@ -137,7 +137,25 @@ api.get = function(req, res, next) {
   q.exec(function(err, group){
     if (err) return next(err);
     if (!group && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
-    res.json(group);
+
+    //Since we have a limit on how many members are populate to the group, we want to make sure the user is always in the group
+    var userInGroup = _.find(group.members, function(member){ return member._id == user._id; });
+
+    //If the group is private or the group is a party, then the user must be a member of the group based on access restrictions above
+    if (group.privacy === 'private' || gid === 'party') {
+      //If the user is not in the group query, add them
+      if (userInGroup === undefined) { group.members.push(user); }
```

Comment:
I'd still argue for `!userInGroup`.

If lodash ever changes how they implement that method, or if we choose to switch methods for calculating that check, `!userInGroup` will still be valid. `userInGroup === undefined` may not be.

Personally, the only time I check for a specific falsy value is if it means something specific to be that particular value. We have another function where it means something different if the value is `null` versus `undefined` and the value can be an empty string. There it makes sense, and is more meaningful to check for a specific kind of falsy value.

Here, I'd argue, it's more meaningful to check if the value is generally falsy, so `!userInGroup`.

Thoughts?